beacon_node/eth2-libp2p/src/behaviour/handler/mod.rs

```diff
@@ -116,26 +140,40 @@ impl<TSpec: EthSpec> ProtocolsHandler for BehaviourHandler<TSpec> {
             Self::Error,
         >,
     > {
+        // TODO: is there any better place to check this?
```

Comment:
Yes. Put this logic in the `connection_keep_alive()` function. This function gets checked periodically.